model/change_point_detection.go

```diff
@@ -428,3 +433,44 @@ func createChangePoint(ctx context.Context, env cedar.Environment, resultToUpdat
 	_, err := env.GetDB().Collection(perfResultCollection).UpdateOne(ctx, filter, update)
 	return errors.Wrap(err, "Unable to create change point")
 }
+
+func TriageChangePoints(ctx context.Context, env cedar.Environment, changePoints map[string]string, status TriageStatus) error {
+	session, err := env.GetDB().Client().StartSession()
+	if err != nil {
+		return errors.Wrap(err, "Unable to start session for triage update")
+	}
+	defer session.EndSession(ctx)
+	if err := session.StartTransaction(); err != nil {
+		return errors.Wrap(err, "Unable to start transaction for triage update")
+	}
+	return mongo.WithSession(ctx, session, func(sc mongo.SessionContext) error {
+		for perfResultId, measurement := range changePoints {
+			filter := bson.M{
+				perfIDKey: perfResultId,
+				bsonutil.GetDottedKeyName(perfAnalysisKey, perfAnalysisChangePointsKey, perfChangePointMeasurementKey): measurement,
+			}
+			update := bson.M{
+				"$set": bson.M{
```

Comment:
The update is the same for all documents, so I think you could do this with a single update command without a transaction right?

Reply:
I think a transaction is needed in order to abort if we don't modify everything we expected to find. Additionally, it seems like reporting 404 for a miss is harder if we don't do these individually.

Reply:
That's reasonable, but I just wanted to double check - the only way for each update to fail is if that measurement variable that is passed in does not exist for the ID passed in. Under what scenarios would that happen?